ports/stm32/spi.c

```diff
@@ -341,6 +363,41 @@ void spi_init(const spi_t *self, bool enable_nss_pin) {
     // an initialisation the next time we use it.
     dma_invalidate_channel(self->tx_dma_descr);
     dma_invalidate_channel(self->rx_dma_descr);
+
+    #if defined(STM32H7)
+    if (0) {
+    #if defined(MICROPY_HW_SPI1_SCK)
+    } else if (spi->Instance == SPI1) {
+        NVIC_SetPriority(SPI1_IRQn, IRQ_PRI_SPI);
```

Comment:
Like in can.c, I think it'd be cleaner to have a variable `irqn` which is set in the big switch above and then here you just need:
```c
#if defined(STM32H7)
NVIC_SetPriority(irqn, IRQ_PRI_SPI);
HAL_NVIC_EnableIRQ(irqn);
#else 
(void)irqn;
#endif 
```